frontend/src/queries/schema.ts

```diff
@@ -757,14 +768,29 @@ export type InsightFilter =
 
 export interface InsightPersonsQuery {
     kind: NodeKind.InsightPersonsQuery
-    source: InsightQueryNode
+    source: InsightQueryNode | RetentionAppearanceQuery
     day?: string
     status?: string
     // TODO: add breakdowns
     // TODO: add fields for other insights (funnels dropdown, compare_previous choice, etc)
     response?: PersonsQueryResponse
 }
 
+export interface PaginatableQuery {
+    /** @asType integer */
+    limit?: number
+    /** @asType integer */
+    offset?: number
+}
+
+export interface RetentionAppearanceQuery extends PaginatableQuery {
+    kind: NodeKind.RetentionAppearanceQuery
+    source: RetentionQuery
+    /** @asType integer */
+    selectedInterval: number
```

Comment:
I think I would rather have the selected interval be part of the retention query, meaning you can get the persons off of it only when this is provided. Otherwise we could end up adding too much unrelated things to persons query that are more the concern of the source query.